docs/proposals/improve-status-in-CAPI-resources.md

```diff
@@ -0,0 +1,921 @@
+those differences are also making it harder for ecosystem tools to build on top of Cluster API, and in some cases
+even confusing new (and old) contributors.
+
+With this proposal Cluster API will close the gap with K8s API conventions in regard to:
+- Polarity: Condition type names should make sense for humans; neither positive nor negative polarity can be recommended
+  as a general rule (already implemented by [#10550](https://github.com/kubernetes-sigs/cluster-api/pull/10550))
+- Use of the Reason field is required (currently in Cluster API reasons is added only when condition are false)
+- Controllers should apply their conditions to a resource the first time they visit the resource, even if the status is Unknown.
+  (currently Cluster API controllers add conditions at different stages of the reconcile loops)
+- Cluster API is also dropping its own Condition type and start using metav1.Conditions from the Kubernetes API.
+
+The last point have also another implication, which is the removal of the Severity field which is currently used
+to determine priority when merging conditions.
+
+TODO Document how we are going to replace severity (currently prototyping)
+
+### Changes to Machine resource
+
+#### Machine Status
+
+Following changes are implemented to Machine's status:
+
+- Disambiguate usage of ready term by renaming fields used for the provisioning workflow
+- Align to K8s API conventions by deprecating `Phase` and corresponding `LastUpdated`
+- Remove `FailureReason` and `FailureMessage` to get rid of the confusing concept of terminal failures
+- Transition to new, improved, K8s API conventions aligned conditions
+
+Below you can find the relevant fields in Machine Status v1beta2, after v1beta1 removal (end state);
+After golang types, you can find a summary table that also shows how changes will be rolled out according to K8s deprecation rules.
+
+```golang
+type MachineStatus struct {
+
+    // Initialization provides observations of the Machine initialization process.
+    // NOTE: fields in this struct are part of the Cluster API contract and are used to orchestrate initial Machine provisioning.
+    // The value of those fields is never updated after provisioning is completed.
+    // Use conditions to monitor the operational state of the Machine's BootstrapSecret.
+    // +optional
+    Initialization *MachineInitializationStatus `json:"initialization,omitempty"`
+    
+    // Represents the observations of a Machine's current state.
+    // +listType=map
+    // +listMapKey=type
+    Conditions []metav1.Condition `json:"conditions,omitempty"`
+    
+    // Other fields...
+    // NOTE: `Phase`, `LastUpdated`, `FailureReason`, `FailureMessage` fields won't be there anymore
+}
+
+// MachineInitializationStatus provides observations of the Machine initialization process.
+type MachineInitializationStatus struct {
```

Comment:
Pointed this out above, but how are we going to provide some feedback in `kubectl get machines`?

Reply:
this will surface in print columns because the machine won't be ready (pending the discussion on status)

Reply:
resolving because I'm consolidating all the discussion about status.phase into https://github.com/kubernetes-sigs/cluster-api/pull/10897#discussion_r1685919394